### ISSUE TITLE
Projecting on a Basis with a combination of ElementVector and ElementComposite

### ISSUE THREAD

**Issue by HelgeGehring** [kinnala/scikit-fem#975]
```python
from skfem import *
from skfem.helpers import *

mesh = MeshTri()

basis = Basis(mesh, ElementVector(ElementTriP1()) * ElementTriP1())


@LinearForm
def from_form(v_t, v_z, w):  # e_t=[y,-x], e_z = 2
    return w.x[1] * v_t[0] - w.x[0] * v_t[1] + 2 * v_z


@BilinearForm
def to_form(e_t, e_z, v_t, v_z, w):
    return inner(e_t, v_t) + inner(e_z, v_z)


x_0 = solve(to_form.assemble(basis), from_form.assemble(basis))
x_1 = basis.project(lambda x: [np.array((x[1], -x[0])), 2])
```
It seems that basis.project doesn't work if we project on a Basis which has a combination of ElementVector and ElementComposite. Or am I doing definition of the lambda wrong?

**Comment by kinnala** [comment]
There is some issue with the logic in _normalize_interp when the output is a list instead of tuple. I'll try to add some tests and fix it.

**Comment by kinnala** [comment]
Now the issue should be fixed in #976 and also be more general, e.g., allow arbitrarily mixing lambdas and constants.

**Comment by HelgeGehring** [comment]
Sounds amazing!
